### PR TITLE
Add sponsorship landing page for SaaS

### PR DESCRIPTION
**Pull request by ericholscher**
This is a first attempt at framing a SaaS/Sponsorship product. It's
targeted mainly at developers who might know what Open Collective is, but
we could make it more generic if we wanted.

This also has a few small refactors that I noticed while doing this, and
wanting to fix a couple typos.

## Preview
127 0 0 1_8000_saas_sponsorship_ (1)

### CODE REVIEW

File: content/pages/landing-pages/sponsorship.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+title: Hosted sponsorship platform for your open source project
+slug: saas
+url: saas/sponsorship/
```
Comment (davidfischer) [inline content/pages/landing-pages/sponsorship.md:3]: Should we maybe simplify this URL? Having `/saas/sponsorship/` when `/saas/` is a 404 feels weird. Maybe just `/sponsorship-platform/` or the like?

**Review by davidfischer — APPROVED** [review]
This looks good. Just to clarify, this isn't linked from anywhere. The plan is to just have a page we can send to interested parties?

File: content/pages/landing-pages/sponsorship.md (hunk 1 above)
Comment (ericholscher) [reply content/pages/landing-pages/sponsorship.md:3]: I was just assuming that we'd have `/saas/` at some point, but that seems fine too.

**Comment by ericholscher** [comment]
> This looks good. Just to clarify, this isn't linked from anywhere. The plan is to just have a page we can send to interested parties?

@davidfischer Good point. I think we should link it somewhere so it gets indexed. Will add it somewhere for now.

**Pull request merged**